var/spack/repos/builtin/packages/hipblas/package.py

```diff
@@ -127,11 +128,13 @@ def cmake_args(self):
             args.append(self.define("BUILD_FILE_REORG_BACKWARD_COMPATIBILITY", True))
         if self.spec.satisfies("@5.3.0:"):
             args.append("-DCMAKE_INSTALL_LIBDIR=lib")
+        if self.spec.satisfies("@6.1:") and self.run_tests:
+            args.append(self.define("LINK_BLIS", "OFF"))
 
         return args
 
-    @run_after("build")
-    @on_package_attributes(run_tests=True)
-    def check_build(self):
-        exe = Executable(join_path(self.build_directory, "clients", "staging", "hipblas-test"))
+    def check(self):
```

Comment:
Just so I'm sure, your goal here *is* to override the existing `check` method (https://spack.readthedocs.io/en/latest/packaging_guide.html#id27), which would call, if available, `make check` and`make test`, right?

Reply:
I ask because it isn't clear from the description what the failure being fixed is.

Reply:
Also, the failure is from the missing dependencies and the cmake arg I added, so keeping the check_build function or overriding the check function doesn't make a difference in terms of the actual test running.

Reply:
> Yes I want to overide the check method since make check and make test aren't available. 

Since they aren't available then by all means override the `check` method.

>  Doing it this way also outputs the test results into a seperate install-time-test-log.txt file. 

That is a handy file to check the outputs.

> Is there a preference for post-build tests on whether I should overide the check function or have the check_build function run after build?

None that I am aware.  Since you don't have `make check` or `make test`, then overriding seems like the right way to go IMO.